File: EJER_25/Referencia.py
from os import system

class Referencia:
    
    def __init__(self,x):
        self.xi = x
        
    def incremento(self):
        self.xi += 1
        return self

def meta():
    print("Traducir el ejercicio 24 de java a Python")

def calculos():
    r = Referencia(0)
    alias = Referencia(0)
    
    alias = r.incremento().incremento().incremento()
    alias.incremento().incremento()
    
    return r,alias

def salida(resulR,resulA):
    print("r = "+str(resulR.xi))
    print("alias = "+str(resulA.xi))

def main():
    system("cls")
    meta()
    rR,rA = calculos()
    salida(rR,rA)

main()

File: 20A40/cir-Israel/Ejercicio 24/Referencia.java
class Referencia {  //tomado de Bruce Eckel  
  int x;         // Thinking in java    
    
Referencia incremento() {    
  x++;    
  return this;
}
public static void main(String []args) {   
  Referencia r= new Referencia();
  Referencia alias; // es null

  alias = r.incremento().incremento().incremento();
  alias.incremento().incremento();   
  System.out.println("x="+ r.x);
  System.out.print("alias="+ alias.x);  
 }
}